### DISCUSSION TITLE
Different email templates for email verification (register vs change email)

### DISCUSSION THREAD

**Discussion by Quasarman** [ory/kratos#2061] [Q&A] [answered]
Hello,

I am not sure it this is possible but since we have access to the identity inside the email templates with 8.1 I was wondering if it is possible to differentiate between users just registering for the first time and users who receive a verification email because they changed it via the settings flow? The reason behind my question is I want to display different messages like:

```
{{- if eq .Identity.just_registered false -}}
{{ template "email.body.welcome.gotmpl" . }}
{{- else -}}
{{ template "email.body.changed_email.gotmpl" . }}
{{- end -}}
```

I know that `just_registered` is not a thing but I was wondering it I somehow can differentiate between those two cases. I guess it works if this is not possible, but displaying a welcome message again after a user changes their email kind of odd.

**Comment by vinckr** [comment]
Welcome @Quasarman!
I have not tested this but just had an idea,
you could check if
```
    "created_at": "2021-12-14T13:37:42.39104Z",
    "updated_at": "2021-12-14T13:37:42.39104Z"
```
are the same, and if true display the `just_registered` message.
Do you think that could work?

**Comment by Quasarman** [comment]
Hello @vinckr,

Thanks for the quick replay. This could work, I just have to check when the `updated_at` field is set, before or after the email was sent.
Question is if I should treat the values as strings when comparing them or somehow transform them to actual timestamps in order to compare a range...
But thanks for your input I'll mark your comment as the answers (since it can take some time since I come around to actually implement it)

**Reply by vinckr** [reply]
Ok! 
I will also see if I can find out how others solved it and if there is a "best practice". When we find a good solution we should document it :-)

**Reply by Quasarman** [reply]
Agreed! I think this is useful in may cases that involve email verification/registration flows.

**Comment by Quasarman** [comment]
Hey I just had the time to try this but sadly it is not working. It seems that comparing `updated_at` and `created_at` always results in true, thus only the "Welcome" message is shown.

```
{{- if eq .Identity.created_at .Identity.updated_at -}}
  <h1>Welcome</h1>
{{- else -}}
  <h1>Email Changed</h1>
{{- end -}}
```

I had a look at the identity object and indeed the `created_at` and `updated_at` values are identical even after changing the email address multiple time. I tried it with v0.8.2-alpha.1

**Comment by Quasarman** [comment, marked as answer]
Any other suggestions? Is this maybe a bug that I shall report. Because that the created_at and updated_at times stay the same seems odd to me.

**Reply by aeneasr** [reply]
Yes definitely, we could probably add some context to the verification step to include which flow you're in (registration vs settings)

**Reply by Quasarman** [reply]
Ok will open a issue tomorrow after work.

**Reply by vinckr** [reply]
Hello @Quasarman,
are you still facing this issue, have you opened an issue for this feature? If yes please link it here 🙏 
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝

**Reply by Quasarman** [reply]
Hello @vinckr 

Yes i have opened the issues its #2480.

BR,
Johannes
